### PR TITLE
[WIP] Use dprint for internal js/ts formating

### LINKED ISSUE #4307: Migrate from prettier to dprint for internal formatting
Currently we use the Rust parts of [dprint](https://docs.rs/dprint/0.1.0-alpha/dprint/) to power `deno fmt`, but we utilise [prettier](https://prettier.io/) for formatting as part of our toolchain.

Is suspect this means we would need to create a standalone minimalistic binary that wraps dprint to do the file system bindings.

cc/ @dsherret

### PR DESCRIPTION
**Pull request by keroxp**
- fixes #4307 
---
- [x] Dprint - Use `cargo run fmt` with latest `deno` build
- [x] Code adaption for dprint bug
  - All error syntaxes (for dprint) have been fixed. 
- [ ] Excluded files -- currently there's no way to exclude specified files for `deno fmt`
  - `.prettierignore` doesn't work
- [ ] Build for per formatting - Because cargo run will re-build entire project...
- [ ] Huge stdout
  - `cargo run` shows all expanded glob arguments...

### CODE REVIEW

**Comment by bartlomieju** [comment]
Hey @keroxp I'm very glad you're tackled this issue!

Just yesterday @dsherret merged PR to add Rust shell to dprint (https://github.com/dsherret/dprint/pull/162), and it's now available on https://crates.io/crates/dprint. 
That means you should be able to `cargo install dprint` just like for `rustfmt`. It's a simple shell that works almost the same as `deno fmt`, the only difference is that `dprint` doesn't walk directory tree to find files, but requires to pass all of them as CLI arguments.

Keep in mind that there are still [some quirks for Deno codebase](https://github.com/dsherret/dprint/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22Changes+for+Deno+Migration%22) when using dprint.

**Comment by kitsonk** [comment]
Also, although this is great, we should merge #4498 first as it includes a move to ES private fields which was blocked because of a lack of Prettier support.  I "could" unpick it, but it is complex.

I also agree that we should use the standalone binary, instead of another version of Deno.

**Comment by bartlomieju** [comment]
> Also, although this is great, we should merge #4498 first as it includes a move to ES private fields which was blocked because of a lack of Prettier support. I "could" unpick it, but it is complex.
> 
> I also agree that we should use the standalone binary, instead of another version of Deno.

@kitsonk, yeah agreed #4498 should be landed first

File: cli/js/net.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export class ListenerImpl implements Listener {
   async *[Symbol.asyncIterator](): AsyncIterator<Conn> {
     while (true) {
       try {
-        yield await this.accept();
+        yield (await this.accept());
```
Comment (dsherret) [inline cli/js/net.ts]: This was fixed in swc so is not necessary anymore.

**Comment by dsherret** [comment]
Before you actually run it on the code, could we wait until this milestone is resolved in dprint? https://github.com/dsherret/dprint/milestone/2

It's way less problems than before, but still a bit of work to do. Might be two more weeks or so :)

**Comment by keroxp** [comment]
@dsherret Yes, this may be too early adaption to dprint. I'll re-try after that.

**Pull request closed without merging**